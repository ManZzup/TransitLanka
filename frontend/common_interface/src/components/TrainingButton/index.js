import React from 'react';

const TrainingButton = (props) => (
  <div className="row">
    {!props.loadButton &&
      <input type="button" className="button warning" value="New Training Set" onClick={props.onSearch} />
    }
    {props.loadButton &&
      <span href="javascript:void(0)" onClick={() => {return false}} className="button warning">
        <img src="/assets/gear.gif" width="40px" alt="" />
      </span>
    }
  </div>
);

export default TrainingButton;
