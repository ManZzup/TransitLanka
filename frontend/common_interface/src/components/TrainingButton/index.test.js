import React from 'react';
import TrainingButton from './';
import renderer from 'react-test-renderer';

it('renders training button correctly', () => {
  const tree = renderer.create(
    <TrainingButton loadButton={false} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders loading button correctly', () => {
  const tree = renderer.create(
    <TrainingButton loadButton={true} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
