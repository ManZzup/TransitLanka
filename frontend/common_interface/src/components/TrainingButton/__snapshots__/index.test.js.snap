exports[`test renders loading button correctly 1`] = `
<div
  className="row">
  <span
    className="button warning"
    href="javascript:void(0)"
    onClick={[Function]}>
    <img
      alt=""
      src="/assets/gear.gif"
      width="40px" />
  </span>
</div>
`;

exports[`test renders training button correctly 1`] = `
<div
  className="row">
  <input
    className="button warning"
    onClick={undefined}
    type="button"
    value="New Training Set" />
</div>
`;
